certbot-apache/certbot_apache/dualparser.py

```diff
@@ -18,10 +18,62 @@ def __getattr__(self, aname):
         """ Attribute value assertion """
         firstval = getattr(self.primary, aname)
         secondval = getattr(self.secondary, aname)
-        if not callable(firstval):
-            assertions.assertEqualSimple(firstval, secondval)
+        exclusions = [
+            # Metadata will inherently be different, as ApacheParserNode does
+            # not have Augeas paths and so on.
+            aname == "metadata",
+            callable(firstval)
+        ]
+        if not any(exclusions):
+            if not isinstance(firstval, bool):
+                # Assertion check does not work for bool like enabled
```

Comment:
I don't see the problem here? If it's set to `PASS`, it won't be a boolean. Unless we're not setting booleans to `PASS`? Is this breaking a test somewhere?

Reply:
If it's a boolean value, like for `enabled` we don't know if we should let the single value to pass assertion or not. At this point we don't know if the value is set to PASS, and furthermore `anyothervalue` should pass assertion against another `anyothervalue` so we should be testing them regardless of being strings.

Reply:
Ah sorry meant to bring this up in discussion earlier -- still not quite sure I understand. Maybe an example might help?

Reply:
Sure, no problem. For most of the values, we want to handle the assertion as per `assertions.assertEqualSimple()` as its able to either pass `assert` for matching values, raise `AssertionError` for mismatches or omit the `assert` if one of the values has `PASS` in it. This works for strings and sequences: 
 - `"a_value"` vs. `"a_value"` - this should pass the `assert`
 - `"a_value"` vs. `"mismatch"` - this should raise `AssertionError`
 - `"a_value"` vs. `"PASS"` - should omit the assert, and not raise `AssertionError`
 - `["a_value", "another_value"]` vs. `["PASS"]` - should omit the assert, and not raise `AssertionError`

For booleans however, we have no means to signal to `assertEqualSimple()` that we intend to omit the `assert`, as the value might not be implemented yet. Furthermore, `assertions.isPass()` that gets called by `assertions.assertEqualSimple()` does `return PASS in value`, which obviously doesn't work for booleans.

That said, we could handle omitting the checks for booleans in `assertions.assertEqualSimple()` as well.

Reply:
> For booleans however, we have no means to signal to `assertEqualSimple()` that we intend to omit the `assert`, as the value might not be implemented yet. 

This is the part that I'm getting hung up on. What do you mean by "as the value might not be implemented yet"? Isn't that what setting the value to `PASS` means?

That being said, `return PASS in value` not working for booleans is a great point, and I also do think I would be more comfortable omitting the check in `assertions.assertEqualSimple()`.

Reply:
> This is the part that I'm getting hung up on. What do you mean by "as the value might not be implemented yet"? Isn't that what setting the value to PASS means?

Yeah, however, if we had `Implementation_A` that had the `enabled` variable implemented and working, and `Implementation_B` which would set the (expected boolean value) to `PASS`, the check would raise a `TypeError` for `Implementation_A`.

Reply:
That's just because of the `PASS in value` bit though, right? We could fix that.

Reply:
Yeah, that's the only reason, however we're still unable to signal the potential `assert` if we want it  to pass or not, so I'd probably skip changing the `isPass()` . When we have everything implemented and working in both parsers, we might revisit this decision and change it.